platform/commonUI/browse/src/ObjectHeaderController.js

```diff
@@ -42,23 +42,37 @@ define(
          * @param event the mouse event
          */
         ObjectHeaderController.prototype.updateName = function (event) {
-            if (event && (event.type === 'blur' || event.which === 13)) {
-                var name = event.currentTarget.innerHTML;
+            if (!event || !event.currentTarget) {
+                return;
+            }
 
-                if (name.length === 0) {
-                    name = "Unnamed " + this.domainObject.getCapability("type").typeDef.name;
-                    event.currentTarget.innerHTML = name;
-                }
+            if (event.type === 'blur') {
+                this.updateModel(event);
+            } else if (event.which === 13) {
+                this.updateModel(event);
+                event.currentTarget.blur();
+                window.getSelection().removeAllRanges();
+            }
+        };
 
-                if (name !== this.$scope.domainObject.model.name) {
-                    this.domainObject.getCapability('mutation').mutate(function (model) {
-                        model.name = name;
-                    });
-                }
+        /**
+         * Updates the model.
+         *
+         * @param event the mouse event
+         * @param private
+         */
+        ObjectHeaderController.prototype.updateModel = function (event) {
+            var name = event.currentTarget.textContent.replace(/\n/g, ' ');
+
+            if (name.length === 0) {
+                name = "Unnamed " + this.domainObject.getCapability("type").typeDef.name;
+                event.currentTarget.textContent = name;
+            }
 
-                if (event.which === 13) {
-                    event.currentTarget.blur();
-                }
+            if (name !== this.domainObject.model.name) {
```

Comment:
Should use `domainObject.getModel().name`. `model` is an undocumented field and may not be present in all circumstances.